integration/mesos/src/main/java/alluxio/mesos/AttributeEvaluator.java

```diff
@@ -0,0 +1,332 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.mesos;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.StreamTokenizer;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.mesos.Protos;
+
+/**
+ * Boolean expression evaluator for Mesos slave attributes
+ */
+public class AttributeEvaluator {
+
+  private Map<String, String> mAttrsMap = new HashMap<String, String>();
+
+  private void setAttributes(List<Protos.Attribute> attributes) {
+    if (attributes == null || attributes.isEmpty()) {
+      return;
+    }
+
+    for (Protos.Attribute attr : attributes) {
+      // Mesos attribute : text ":" ( scalar | range | text )
+      // now only support text attributes.
+      mAttrsMap.put(attr.getName(), attr.getText().getValue());
+    }
+  }
+
+  /**
+   * @param attributes Mesos slave attributes
+   */
+  public AttributeEvaluator(List<Protos.Attribute> attributes) {
+    setAttributes(attributes);
+  }
+
+  /**
+   * @param constraints Mesos slave attributes
+   * @return return True if matched constraints
+   */
+  public boolean matchAttributes(String constraints) {
```

Comment:
Should this return RunTimeException if the parsing fails? Or some checked exception?